core/src/main/java/com/linecorp/armeria/server/file/HttpDataFile.java

```diff
@@ -73,23 +76,23 @@ protected String pathOrUri() {
     }
 
     @Override
-    public HttpFileAttributes readAttributes() {
+    public HttpFileAttributes attributes() {
         return attrs;
     }
 
     @Override
     public CompletableFuture<HttpFileAttributes> readAttributes(Executor fileReadExecutor) {
-        return AggregatedHttpFile.super.readAttributes(fileReadExecutor);
+        return UnmodifiableFuture.completedFuture(attrs);
```

Comment:
Is there any reason that you didn't use `CompletableFuture.completeValue(attrs)`?

Reply:
No, there isn't any reason. Should we?

Reply:
I prefer to use `UnmodifiableFuture` only when the future can be complete by the caller by mistake, but no big deal. :)

Reply:
A user can also override the result by using `obtrude*()` methods, so I wanted to avoid such situation. Let me leave this as it is and reconsider once we find an issue about it.